core/math/bvh.h

```diff
@@ -281,16 +281,17 @@ class BVH_Manager {
 	}
 
 	// prefer calling this directly as type safe
-	void set_pairable(const BVHHandle &p_handle, bool p_pairable, uint32_t p_pairable_type, uint32_t p_pairable_mask) {
-		tree.item_set_pairable(p_handle, p_pairable, p_pairable_type, p_pairable_mask);
+	void set_pairable(const BVHHandle &p_handle, bool p_pairable, uint32_t p_pairable_type, uint32_t p_pairable_mask, bool p_force_collision_check = true) {
+		// Returns true if the pairing state has changed and the item is active.
+		bool needs_update = tree.item_set_pairable(p_handle, p_pairable, p_pairable_type, p_pairable_mask);
 
 		if (USE_PAIRS) {
 
 			// not sure if absolutely necessary to flush collisions here. It will cost performance to, instead
 			// of waiting for update, so only uncomment this if there are bugs.
 			//_check_for_collisions();
 
-			if (get_active(p_handle)) {
+			if (needs_update || (p_force_collision_check && tree.item_get_active(p_handle))) {
 				// when the pairable state changes, we need to force a collision check because newly pairable
```

Comment:
This can use the `get_active(p_handle)` shortcut (if it is there we might as well use it for less verbosity, and encapsulation).

Reply:
In fact I suspect for clarity it may be worth keeping the `get_active` check here in both cases, and having the bool returned be a `something_changed`. Unless profiling shows it to be the case, I don't think combining the two in one bool gains much, at the expensive of lowered clarity.